components/TransactionForm.js

```diff
@@ -1,26 +1,34 @@
 import { categories } from "@/lib/categories";
+import { transactions } from "@/lib/transactions";
 import { useState } from "react";
 import CurrencyInput from "react-currency-input-field";
 import styled from "styled-components";
 
-export default function TransactionForm({ onAdd }) {
+export default function TransactionForm({
+  onAdd,
+  formHeader,
+  buttonText,
+  updatedData = {},
+  isEditing,
+}) {
   const today = new Date().toISOString().split("T")[0];
-
-  const [selectedType, setSelectedType] = useState("");
-  const [amount, setAmount] = useState("");
+  const [selectedType, setSelectedType] = useState(updatedData.type || "");
+  const [amount, setAmount] = useState(updatedData.amount?.toString() || "");
   const [typeError, setTypeError] = useState(false);
+  const [amountError, setAmountError] = useState(false);
 
   function handleSubmit(event) {
     event.preventDefault();
-
     const formData = new FormData(event.target);
+
     const data = Object.fromEntries(formData);
 
-    if (data.amount) {
-      data.amount = parseFloat(
-        data.amount.replace(/\./g, "").replace(",", ".")
-      );
-    }
+    data.amount = parseFloat(amount.replace(/\./g, "").replace(",", "."));
+
+    if (data.amount <= 0) {
+      setAmountError(true);
+      return;
+    } else setAmountError(false);
```

Comment:
if you can make sure that amount is always positive, you don't need to check this.